File: benchmark_scripts/connected_components/python/connected_components.py
import sys
import time
from scipy.io import mmread
from scipy.sparse import coo_array

def cc(filename, maxi=100):
    G = mmread(filename)
    start = time.time()
    c = coo_array(list(map(lambda i: float(i), range(1, G.shape[0] + 1, 1))))

    for iter in range(maxi):
        x = G.multiply(c.transpose()).max(axis=0)
        c = c.maximum(x)
    end = time.time()
    print(end - start)

if __name__ == "__main__":
    args = sys.argv
    assert len(args) == 2
    filename = args[1]
    cc(filename)

File: benchmark_scripts/connected_components/jl/connected_components.jl
using MatrixMarket
using SparseArrays

function cc(filename, maxi)
  G = MatrixMarket.mmread(filename)
  start = time()
  c = vec(collect(1.0:1.0:float(size(G, 1))))

  for iter in 1:maxi
    x = maximum(G .* c, dims=1)
    c = max.(c, transpose(x))
  end
  fin = time()
  println(fin - start)
end

@assert(length(ARGS) == 1)
filename = ARGS[1]
maxi = 100
cc(filename, maxi)

